ovos_bus_client/session.py

```diff
@@ -215,17 +240,26 @@ def get(message=None):
         sess = SessionManager.default_session
         message = message or dig_for_message()
 
+        # A message exists, get a real session
+        if message:
+            LOG.info(f"Check for session in message")
+            msg_sess = Session.from_message(message)
+            if msg_sess.session_id in SessionManager.sessions:
+                SessionManager.sessions[msg_sess.session_id] = msg_sess
+                return msg_sess
```

Comment:
should this check be here and return?

if the session is not in sessions it should be added there (it's a new session from the client)

```python
msg_sess = Session.from_message(message)
# below should happen unconditionally
SessionManager.sessions[msg_sess.session_id] = msg_sess
```

Reply:
I was assuming this was a new/invalid session in this case.. perhaps `from_message` should have an optional param so we can call it without creating a new session?

Reply:
sessions are never invalid, they may just be untracked... by definition if you get it in message you should start tracking it, or update its timestamp

Reply:
But if we create a new one for a GUI message, etc. we don't want to track it, right? (or maybe not even create it?)

Reply:
also to clarify expiration, a session never truly expires since they come from client side, you could always resume a previous conversation from anytime long ago if you send it with a new utterance message.

the expiration is when core should stop tracking it, doesnt mean it is invalid when we see it

Reply:
> But if we create a new one for a GUI message, etc. we don't want to track it, right? (or maybe not even create it?)

you shouldnt create one in this case.... you can if you want for some use case, but by default this should just assign default session in use. a GUI message is emited as part of some session... if it doesnt know about it we assign it and it can be tracked in responses. this is client side responsability, a sort of memory

Reply:
Right, so for a message without session data that is passed to `Session.from_message`, we would want a flag to prevent creating a new session in those cases

Reply:
i see the confusion now, from_message should not create a new session!

the bug is that where you added the warning log it should actually be assigning the default session